collector/pg_stat_wal.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package collector
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log/slog"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const statWALSubsystem = "stat_wal"
+
+func init() {
+	registerCollector(statWALSubsystem, defaultDisabled, NewPGStatWALCollector)
+}
+
+type PGStatWALCollector struct {
+	log *slog.Logger
+}
+
+func NewPGStatWALCollector(config collectorConfig) (Collector, error) {
+	return &PGStatWALCollector{log: config.logger}, nil
+}
+
+var statsWALRecordsDesc = prometheus.NewDesc(
+	prometheus.BuildFQName(namespace, statWALSubsystem, "wal_records"),
```

Comment:
Counter values must end in `_total`

```suggestion
	prometheus.BuildFQName(namespace, statWALSubsystem, "wal_records_tota"),
```